dashboard/src/modules/components/OverviewComponent/common-component.jsx

```diff
@@ -157,3 +171,164 @@ export const RenderPagination = (props) => {
     />
   );
 };
+
+export const EditRow = (props) => {
+  return (
+    <div className="pf-c-inline-edit__action pf-m-enable-editable">
+      {!props.item.isEdit ? (
+        <Button
+          variant="plain"
+          onClick={() => props.toggleEdit(props.item.resource_id, true)}
+          icon={<PencilAltIcon />}
+        />
+      ) : (
+        <div>
+          <Button
+            isDisabled={
+              !props.item.isDirty ||
+              !props.item.name ||
+              props.item.name_validated === "error"
+            }
+            onClick={() =>
+              props.saveRowData("datasetName", props.item, props.item.name)
+            }
+            variant="plain"
+            icon={<CheckIcon />}
+          />
+          <Button
+            variant="plain"
+            onClick={() => props.toggleEdit(props.item.resource_id, false)}
+            icon={<TimesIcon />}
+          />
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const DatasetNameInput = (props) => {
+  return (
+    <TextInput
+      validated={props.validated}
+      value={props.value}
+      type="text"
+      onChange={props.onChange}
+      aria-label="Edit Dataset name"
+    />
+  );
+};
+
+export const NewRunsRow = (props) => {
```

Comment:
Why is this function in this module?  I.e., what is "common" about it?  Why isn't it in the `NewRunsComponent` module?  (Analogously, the `SavedRunsRow` function?)

Reply:
It uses most of the functions in common-components. And, instead of importing those functions in both `NewRunsComponent` and `SavedRunsComponent` I preferred to write `NewRunsRow` and `SavedRunsRow` is common-components file of Overview Component